src/main/java/org/gridsuite/loadflow/server/LoadFlowController.java

```diff
@@ -42,14 +43,16 @@
     public ResponseEntity<LoadFlowResult> loadFlow(@ApiParam(value = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
                                                    @ApiParam(value = "Other networks UUID") @RequestParam(name = "networkUuid", required = false) List<String> otherNetworks,
                                                    @ApiParam(value = "Provider") @RequestParam(name = "provider", required = false) String provider,
+                                                   @ApiParam(value = "reportId") @RequestParam(name = "reportId", required = false) UUID reportId,
+                                                   @ApiParam(value = "reportName") @RequestParam(name = "reportName", required = false) String reportName,
                                                    @RequestBody(required = false) String loadflowParams) {
         LoadFlowParameters parameters = loadflowParams != null
                 ? JsonLoadFlowParameters.read(new ByteArrayInputStream(loadflowParams.getBytes()))
                 : null;
 
         List<UUID> otherNetworksUuid = otherNetworks != null ? otherNetworks.stream().map(UUID::fromString).collect(Collectors.toList()) : Collections.emptyList();
 
-        LoadFlowResult result = loadFlowService.loadFlow(networkUuid, otherNetworksUuid, parameters, provider);
+        LoadFlowResult result = loadFlowService.loadFlow(networkUuid, otherNetworksUuid, parameters, provider, Optional.ofNullable(reportId), Optional.ofNullable(reportName));
```

Comment:
Usually, optional is used to return a value not as an argument

Reply:
Optional as an argument is more that unusual, it is FORBIDDEN !